BaseClasses.py

```diff
@@ -9,12 +9,12 @@
 import typing  # this can go away when Python 3.8 support is dropped
 from argparse import Namespace
 from collections import Counter, deque
-from collections.abc import Collection, MutableSequence
+from collections.abc import Collection, MutableSequence, Sequence
 from enum import IntEnum, IntFlag
 from typing import (AbstractSet, Any, Callable, ClassVar, Dict, Iterable, Iterator, List, Mapping, NamedTuple,
-                    Optional, Protocol, Set, Tuple, Union, Type)
+                    Optional, Protocol, Set, Tuple, Union, Type, TypedDict)
 
-from typing_extensions import NotRequired, TypedDict
+from typing_extensions import NotRequired
```

Comment:
I think this should not change. `TypedDict` has some extra bug fixes, features, and optimizations from `typing_extensions` (backported from the latest Python).